### PR TITLE
Fix string interner size telemetry

### PR DESCRIPTION
**Pull request by scottopell**
### What does this PR do?

Fixes some string interner telemetry by tagging with a unique interner ID.

### Motivation

When running with multiple workers, some of the workers step on each other while incrementing/decrementing the telemetry.
This adds clarity both for how many interners are present and what the status of each individual interner is.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by gh123man — APPROVED** [review]

**Review by blt — APPROVED** [review]

**Comment by scottopell** [comment]
There is more to fix here

File: comp/dogstatsd/server/intern.go [hunk 1]
```diff
@@ -66,30 +69,28 @@ func (i *stringInterner) LoadOrStore(key []byte) string {
 	// See https://github.com/golang/go/commit/f5f5a8b6209f84961687d993b93ea0d397f5d5bf
 	if s, found := i.strings[string(key)]; found {
 		if i.tlmEnabled {
-			tlmSIRHits.Inc()
+			tlmSIRHits.WithTags(map[string]string{"internerId": i.id}).Inc()
```
Comment (remeh) [inline comp/dogstatsd/server/intern.go:72]: Not that it matters much, but we should probably use snake case for the tag, `interner_id`, it'll be more consistant with the rest in graphs.

File: comp/dogstatsd/server/parse.go [hunk 2]
```diff
@@ -50,16 +50,17 @@ type parser struct {
...
 }
 
-func newParser(cfg config.Reader, float64List *float64ListPool) *parser {
+func newParser(cfg config.Reader, float64List *float64ListPool, workerNum int) *parser {
 	stringInternerCacheSize := cfg.GetInt("dogstatsd_string_interner_size")
 	readTimestamps := cfg.GetBool("dogstatsd_no_aggregation_pipeline")
 
-	return &parser{
-		interner:         newStringInterner(stringInternerCacheSize),
+	p := &parser{
```
Comment (remeh) [inline comp/dogstatsd/server/parse.go:57]: Did you use a var for debugging (debugger / to print it out)? You did the same with the worker. Don't bother about my comment if it's just a style preference.

**Review by remeh — APPROVED** [review]
Looks good to me. Make sure to fix all these unit tests before merging 👍

**Pull request merged**